crates/isograph_compiler/src/watch.rs

```diff
@@ -32,16 +35,17 @@ pub async fn handle_watch_command(
             .expect("Failing when watching schema extension");
     }
 
+    let mut state = CompilerState::new(config);
+    compile_and_print(&mut state, &Vec::new());
```

Comment:
`&[]` instead of `&Vec::new()`, but more to the point, `compile_and_print` should take `Option<changes>`.

In addition, it would be nice to use the type system to ensure that `source_files.read_and_parse_all` is called before `validate_and_create_artifacts` is called. (Maybe we can also ensure that `read_and_parse_all` is called before `update`, but that's a little less clear in the language server case.) Maybe we can try:
- `CompilerState.source_files` is `Option<SourceFiles>`, and the only way to make it `Some` is to call `read_and_parse_all`
- When we get to the language server, we can consider a "better" (but more verbose) solution, `enum SourceFileState { PartiallyLoaded(PartiallyLoadedSourceFiles) FullyLoaded(FullyLoadedSourceFiles) }`, where only `FullyLoadedSourceFiles` has a `.validate_and_generate_artifacts` method.